### PR TITLE
Targeting mono

### PR DESCRIPTION
**Pull request by naveensrinivasan**
Targeting mono.

- [x] Build App 
- [x] Run Unit Tests  
- [x] Successfully build it on travis-ci
- [x] Run test's on travis-ci

### CODE REVIEW

**Comment by shiftkey** [comment]
> /usr/lib/mono/xbuild/12.0/bin/Microsoft.Common.targets:  warning : Reference 'NSubstitute, Version=1.8.0.0, Culture=neutral, PublicKeyToken=92dd2e9066daa5ca, processorArchitecture=MSIL' not resolved

Is package restore not working as expected?

**Comment by naveensrinivasan** [comment]
@shiftkey  OSX file system isn't case sensitive which is causing me an issue. So it works on mine but does not work on the linux. 

Took care of `NSubstitute`.The issue was the `NET40` was uppercase instead of `net40` which is the actual case.

**Comment by naveensrinivasan** [comment]
Now I could see this test's failing much more often on mono `StopsMakingNewRequestsWhenTakeIsFulfilled`

**Comment by naveensrinivasan** [comment]
@shiftkey  I am done with these changes.

**Comment by shiftkey** [comment]
@naveensrinivasan thanks, I'll see if that pesky test is a repeatable failure

**Comment by shiftkey** [comment]
And now I'm kinda curious why AppVeyor isn't reporting the status of this PRs...

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+language: csharp
+
+sudo: false  # use the new container-based Travis infrastructure
+os:
+  - osx
+  - linux
+install:
+  - curl -sS http://storage.bos.xamarin.com/bot-provisioning/PortableReferenceAssemblies-2014-04-14.zip > /tmp/pcl-assemblies.zip
+  - unzip /tmp/pcl-assemblies.zip -d /tmp/pcl-assemblies && mv /tmp/pcl-assemblies/PortableReferenceAssemblies-2014-04-14 /tmp/pcl-assemblies/.NETPortable
```
Comment (akoeplinger) [inline .travis.yml]: You shouldn't need this, the PCL assemblies are preinstalled on Travis.

File: Octokit.Tests/Octokit.Tests.csproj [hunk 2]
```diff
@@ -38,7 +38,7 @@
   <ItemGroup>
     <Reference Include="NSubstitute, Version=1.8.0.0, Culture=neutral, PublicKeyToken=92dd2e9066daa5ca, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\packages\NSubstitute.1.8.0.0\lib\net45\NSubstitute.dll</HintPath>
+      <HintPath>../packages/NSubstitute.1.8.0.0/lib/net45/NSubstitute.dll</HintPath>
```
Comment (akoeplinger) [inline Octokit.Tests/Octokit.Tests.csproj]: This shouldn't be required, xbuild converts backslashes to slashes on Unix.

File: build.fsx [hunk 3]
```diff
@@ -94,14 +94,20 @@ let setParams defaults = {
...
         |> DoNothing
 )
 
-Target "BuildXSApp" (fun _ ->
-    build setParams "./Octokit-XamarinStudio.sln"
-        |> DoNothing
+Target "BuildMono" (fun _ ->
+    // xbuild does not support msbuild  tools version 14.0 and that is the reason
+    // for using the xbuild command directly instead of using msbuild
+    Exec "xbuild" "./Octokit-Mono.sln /t:Build /tv:12.0 /v:m  /p:RestorePackages='False' /p:Configuration='Release' /logger:Fake.MsBuildLogger+ErrorLogger,'../octokit.net/tools/FAKE.Core/tools/FakeLib.dll'"
```
Comment (akoeplinger) [inline build.fsx]: Mono 4.2 should support ToolsVersion 14.0, with which Mono did you test it?

**Comment by shiftkey** [comment]
@akoeplinger thanks for the review comments! :heart:

File: build.fsx (hunk 3 above)
Comment (naveensrinivasan) [inline build.fsx]: I tried it and it doesn't work.

>Building project: ./Octokit-Mono.sln
  xbuild  ./Octokit-Mono.sln /t:Build /tv:14.0 /v:m  /p:RestorePackages="False" >/p:Configuration="Release" >/logger:Fake.MsBuildLogger+ErrorLogger,"/home/travis/build/naveensrinivasan/octokit.net/tools/FAKE.Core/tools/FakeLib.dll"
>xbuild  ./Octokit-Mono.sln /t:Build /tv:14.0 /v:m  /p:RestorePackages="False" >/p:Configuration="Release" >/logger:Fake.MsBuildLogger+ErrorLogger,"/home/travis/build/naveensrinivasan/octokit.net/tools/F>>AKE.Core/tools/FakeLib.dll"
>XBuild Engine Version 12.0
Mono, Version 4.2.1.0
>Copyright (C) 2005-2013 Various Mono authors
>MSBUILD: error MSBUILD0000: Unknown tools version: '14.0' . Known versions: '2.0' '3.0' '3.5' '4.0' '12.0'
>Running build failed.
>Error:
>Building ./Octokit-Mono.sln failed with exitcode 1.

https://travis-ci.org/naveensrinivasan/octokit.net/jobs/96853565

File: .travis.yml (hunk 1 above)
Comment (naveensrinivasan) [inline .travis.yml]: I tried this one also. It works on OSX but fails on linux for the same change.

OSX - https://travis-ci.org/naveensrinivasan/octokit.net/jobs/96858329

linux - https://travis-ci.org/naveensrinivasan/octokit.net/jobs/96858330

Comment (akoeplinger) [inline .travis.yml]: I think you meant https://travis-ci.org/naveensrinivasan/octokit.net/jobs/96850533.

If you look at the actual build log, the error there is from something completely unrelated (`System.TypeInitializationException: The type initializer for 'System.Collections.Generic.List'1' threw an exception. ---> System.Threading.ThreadAbortException: ` during tests).

File: build.fsx (hunk 3 above)
Comment (akoeplinger) [inline build.fsx]: Interesting, thanks. Looks like the default `xbuild` script invokes xbuild 12.0, which is probably why this happens. Not sure if that's intended, I'll take a closer look.

File: .travis.yml (hunk 1 above)
Comment (naveensrinivasan) [inline .travis.yml]: But it works on OSX and fails on linux. The same code. Do you know why? 

And when I include PCL it does not fail every time.

Comment (akoeplinger) [inline .travis.yml]: PCL has nothing to do with it, that's just a coincidence (the random ThreadAbortException looks like a Mono bug).

Comment (naveensrinivasan) [inline .travis.yml]: OK. Thanks.

**Comment by naveensrinivasan** [comment]
@shiftkey I am done with changes. There is a conflict on the `README.MD`

**Comment by shiftkey** [comment]
@naveensrinivasan as this is targeting `master`, do you mind merging `master` in to address this conflict and verify the state of the mono build?

**Comment by naveensrinivasan** [comment]
@shiftkey Will do that. Thanks

**Comment by naveensrinivasan** [comment]
@shiftkey  I resolved the conflict. The build failed :tired_face:  in appveyor and linux because of `StopsMakingNewRequestsWhenTakeIsFulfilled`

**Comment by shiftkey** [comment]
@naveensrinivasan ugh, feel free to mute that test and I'll open an issue to investigate

**Comment by naveensrinivasan** [comment]
@shiftkey Thanks! Done!

File: Octokit.Tests/Reactive/ObservableRepositoriesClientTests.cs [hunk 4]
```diff
@@ -90,7 +90,6 @@ public class TheGetAllForCurrentMethod
                 gitHubClient.Connection.Received(1).Get<List<Repository>>(thirdPageUrl, null, null);
             }
 
-            [Fact]
```
Comment (shiftkey) [inline Octokit.Tests/Reactive/ObservableRepositoriesClientTests.cs]: Can you keep this `[Fact]` around with the `Skip` attribute set?

`[Fact(Skip = "See https://github.com/octokit/octokit.net/issues/1011 for issue to investigate this further")]`

File: packages/repositories.config [hunk 5]
```diff
@@ -1,5 +1,10 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <repositories>
+  <repository path="../Octokit.Reactive/packages.config" />
```
Comment (shiftkey) [inline packages/repositories.config]: :lipstick: this just seems to be some mixed up `\` and `/` path separators - can we unify these?

**Comment by naveensrinivasan** [comment]
@shiftkey Got it done! Thanks

File: packages/repositories.config [hunk 6]
```diff
@@ -2,9 +2,14 @@
 <repositories>
   <repository path="..\Octokit.Reactive\packages.config" />
   <repository path="..\Octokit.Tests.Conventions\packages.config" />
+  <repository path="..\Octokit.Tests\packages.Octokit.Tests.config" />
```
Comment (shiftkey) [inline packages/repositories.config]: @naveensrinivasan sorry, I should have been clearer here - these new entries are just duplicates of previous ones - can you :fire: them to simplify the diff?

Comment (shiftkey) [inline packages/repositories.config]: Eh, I'll clean this up after `master` goes green again.

**Comment by shiftkey** [comment]
@naveensrinivasan thanks again, outstanding stuff!

**Pull request merged**